service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -203,6 +204,28 @@ private PolicyEvaluationResult evalPolicySet(final PolicySet policySet, final St
             }
             result = new PolicyEvaluationResult(effect, new ArrayList<>(subjectAttributes),
                     new ArrayList<>(resourceAttributes), resolvedResourceUris);
+        } catch (QueryException qe) {
+            StringBuilder builder = new StringBuilder();
+            result = new PolicyEvaluationResult(Effect.INDETERMINATE);
+            if (qe.getMessage().contains("Graph search failed: traversal limit")) {
+                //The traversal limit is the seventh word in the exception message
+                String traversalLimit = qe.getMessage().split(" ")[6];
+                builder.append("Traversal limit exceeded").append(policySet.getName()).append(" subject:")
+                        .append(subjectIdentifier).append(", Resource URI: ").append(resourceURI);
+                String message = String.format(
+                        "The number of attributes on this resource/subject %s, %s have exceeded the maximum "
+                        + "limit of %s",
+                        resourceURI, subjectIdentifier, traversalLimit);
+                result.setMessage(message);
+            } else {
+                builder = new StringBuilder();
+                builder.append("Exception occured while evaluating the policy set. Policy Set ID:")
+                        .append(policySet.getName()).append(" subject:").append(subjectIdentifier)
+                        .append(", Resource URI: ").append(resourceURI);
+                LOGGER.error(builder.toString(), qe);
+                result = new PolicyEvaluationResult(Effect.INDETERMINATE);
+            }
+            LOGGER.error(builder.toString(), qe);
```

Comment:
This error message parsing and construction of new message is redundant.  Just append the error message from the exception, i.e., 
+                         builder.append("Exception occured while evaluating the policy set. Policy Set ID:")
 +                        .append(policySet.getName()).append(" subject:").append(subjectIdentifier)
 +                        .append(", Resource URI: ").append(resourceURI).**append(qe.getMessage()**);

And return this message in the result.

In fact we don't need to explicitly catch QueryException.  Inside [catch (Throwable e)](https://github.com/predix/acs/blob/2c0d68993678945a4e5e62b808734f849e1c0055/service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java#L229-L238) block below if message is present attach it to the final message and return in the result.